In [149]:
#620150541, 620148643
#Data Acquisition
import pandas as pd
import numpy as np
from sklearn.preprocessing import LabelEncoder
df = pd.read_csv('lung cancer.csv', encoding="latin1")
df.head(10)

GENDER  AGE  SMOKING  YELLOW_FINGERS  ANXIETY  PEER_PRESSURE  \
0      M   69        1               2        2              1   
1      M   74        2               1        1              1   
2      F   59        1               1        1              2   
3      M   63        2               2        2              1   
4      F   63        1               2        1              1   
5      F   75        1               2        1              1   
6      M   52        2               1        1              1   
7      F   51        2               2        2              2   
8      F   68        2               1        2              1   
9      M   53        2               2        2              2   

   CHRONIC DISEASE  FATIGUE   ALLERGY   WHEEZING  ALCOHOL CONSUMING  COUGHING  \
0                1         2         1         2                  2         2   
1                2         2         2         1                  1         1   
2                1         2         1         2                  1         2   
3                1         1         1         1                  2         1   
4                1         1         1         2                  1         2   
5                2         2         2         2                  1         2   
6                1         2         1         2                  2         2   
7                1         2         2         1                  1         1   
8                1         2         1         1                  1         1   
9                2         1         2         1                  2         1   

   SHORTNESS OF BREATH  SWALLOWING DIFFICULTY  CHEST PAIN LUNG_CANCER  
0                    2                      2           2         YES  
1                    2                      2           2         YES  
2                    2                      1           2          NO  
3                    1                      2           2          NO  
4                    2                      1           1          NO  
5                    2                      1           1         YES  
6                    2                      1           2         YES  
7                    2                      2           1         YES  
8                    1                      1           1          NO  
9                    1                      2           2         YES

In [150]:
#There are no missing values
missing_values = df.isnull().sum()
print("Missing Values:\n", missing_values)

Missing Values:
 GENDER                   0
AGE                      0
SMOKING                  0
YELLOW_FINGERS           0
ANXIETY                  0
PEER_PRESSURE            0
CHRONIC DISEASE          0
FATIGUE                  0
ALLERGY                  0
WHEEZING                 0
ALCOHOL CONSUMING        0
COUGHING                 0
SHORTNESS OF BREATH      0
SWALLOWING DIFFICULTY    0
CHEST PAIN               0
LUNG_CANCER              0
dtype: int64


In [151]:
#Data Transformation - Changing the categorical data into numeric
label_encoder = LabelEncoder()
df['GENDER_ENCODED'] = label_encoder.fit_transform(df['GENDER'])
df['LUNG_CANCER_ENCODED'] = label_encoder.fit_transform(df['LUNG_CANCER'])
df.drop(['GENDER', 'LUNG_CANCER'], axis=1, inplace=True)
df.head(10)

#1 is M and YES

AGE  SMOKING  YELLOW_FINGERS  ANXIETY  PEER_PRESSURE  CHRONIC DISEASE  \
0   69        1               2        2              1                1   
1   74        2               1        1              1                2   
2   59        1               1        1              2                1   
3   63        2               2        2              1                1   
4   63        1               2        1              1                1   
5   75        1               2        1              1                2   
6   52        2               1        1              1                1   
7   51        2               2        2              2                1   
8   68        2               1        2              1                1   
9   53        2               2        2              2                2   

   FATIGUE   ALLERGY   WHEEZING  ALCOHOL CONSUMING  COUGHING  \
0         2         1         2                  2         2   
1         2         2         1                  1         1   
2         2         1         2                  1         2   
3         1         1         1                  2         1   
4         1         1         2                  1         2   
5         2         2         2                  1         2   
6         2         1         2                  2         2   
7         2         2         1                  1         1   
8         2         1         1                  1         1   
9         1         2         1                  2         1   

   SHORTNESS OF BREATH  SWALLOWING DIFFICULTY  CHEST PAIN  GENDER_ENCODED  \
0                    2                      2           2               1   
1                    2                      2           2               1   
2                    2                      1           2               0   
3                    1                      2           2               1   
4                    2                      1           1               0   
5                    2                      1           1               0   
6                    2                      1           2               1   
7                    2                      2           1               0   
8                    1                      1           1               0   
9                    1                      2           2               1   

   LUNG_CANCER_ENCODED  
0                    1  
1                    1  
2                    0  
3                    0  
4                    0  
5                    1  
6                    1  
7                    1  
8                    0  
9                    1

In [152]:
#Removing Outliers
z_scores = np.abs((df - df.mean()) / df.std())
outliers = df[(z_scores > 3).any(axis=1)]
print("Outliers:\n", outliers)

df = df[(z_scores < 3).all(axis=1)]
df.head(10)

Outliers:
      AGE  SMOKING  YELLOW_FINGERS  ANXIETY  PEER_PRESSURE  CHRONIC DISEASE  \
22    21        2               1        1              1                2   
238   38        1               2        1              1                2   

     FATIGUE   ALLERGY   WHEEZING  ALCOHOL CONSUMING  COUGHING  \
22          2         2         1                  1         1   
238         2         2         2                  1         2   

     SHORTNESS OF BREATH  SWALLOWING DIFFICULTY  CHEST PAIN  GENDER_ENCODED  \
22                     2                      1           1               0   
238                    2                      1           2               0   

     LUNG_CANCER_ENCODED  
22                     0  
238                    1  


AGE  SMOKING  YELLOW_FINGERS  ANXIETY  PEER_PRESSURE  CHRONIC DISEASE  \
0   69        1               2        2              1                1   
1   74        2               1        1              1                2   
2   59        1               1        1              2                1   
3   63        2               2        2              1                1   
4   63        1               2        1              1                1   
5   75        1               2        1              1                2   
6   52        2               1        1              1                1   
7   51        2               2        2              2                1   
8   68        2               1        2              1                1   
9   53        2               2        2              2                2   

   FATIGUE   ALLERGY   WHEEZING  ALCOHOL CONSUMING  COUGHING  \
0         2         1         2                  2         2   
1         2         2         1                  1         1   
2         2         1         2                  1         2   
3         1         1         1                  2         1   
4         1         1         2                  1         2   
5         2         2         2                  1         2   
6         2         1         2                  2         2   
7         2         2         1                  1         1   
8         2         1         1                  1         1   
9         1         2         1                  2         1   

   SHORTNESS OF BREATH  SWALLOWING DIFFICULTY  CHEST PAIN  GENDER_ENCODED  \
0                    2                      2           2               1   
1                    2                      2           2               1   
2                    2                      1           2               0   
3                    1                      2           2               1   
4                    2                      1           1               0   
5                    2                      1           1               0   
6                    2                      1           2               1   
7                    2                      2           1               0   
8                    1                      1           1               0   
9                    1                      2           2               1   

   LUNG_CANCER_ENCODED  
0                    1  
1                    1  
2                    0  
3                    0  
4                    0  
5                    1  
6                    1  
7                    1  
8                    0  
9                    1

In [153]:
# Remove trailing spaces from column names
df.columns = df.columns.str.strip()

In [154]:
df.describe()

AGE     SMOKING  YELLOW_FINGERS     ANXIETY  PEER_PRESSURE  \
count  307.000000  307.000000      307.000000  307.000000     307.000000   
mean    62.889251    1.563518        1.570033    1.501629       1.504886   
std      7.754871    0.496759        0.495879    0.500814       0.500792   
min     39.000000    1.000000        1.000000    1.000000       1.000000   
25%     58.000000    1.000000        1.000000    1.000000       1.000000   
50%     62.000000    2.000000        2.000000    2.000000       2.000000   
75%     69.000000    2.000000        2.000000    2.000000       2.000000   
max     87.000000    2.000000        2.000000    2.000000       2.000000   

       CHRONIC DISEASE     FATIGUE     ALLERGY    WHEEZING  ALCOHOL CONSUMING  \
count       307.000000  307.000000  307.000000  307.000000         307.000000   
mean          1.501629    1.671010    1.553746    1.557003           1.560261   
std           0.500814    0.470614    0.497915    0.497551           0.497166   
min           1.000000    1.000000    1.000000    1.000000           1.000000   
25%           1.000000    1.000000    1.000000    1.000000           1.000000   
50%           2.000000    2.000000    2.000000    2.000000           2.000000   
75%           2.000000    2.000000    2.000000    2.000000           2.000000   
max           2.000000    2.000000    2.000000    2.000000           2.000000   

         COUGHING  SHORTNESS OF BREATH  SWALLOWING DIFFICULTY  CHEST PAIN  \
count  307.000000           307.000000             307.000000  307.000000   
mean     1.579805             1.638436               1.472313    1.557003   
std      0.494396             0.481238               0.500048    0.497551   
min      1.000000             1.000000               1.000000    1.000000   
25%      1.000000             1.000000               1.000000    1.000000   
50%      2.000000             2.000000               1.000000    2.000000   
75%      2.000000             2.000000               2.000000    2.000000   
max      2.000000             2.000000               2.000000    2.000000   

       GENDER_ENCODED  LUNG_CANCER_ENCODED  
count      307.000000           307.000000  
mean         0.527687             0.876221  
std          0.500048             0.329866  
min          0.000000             0.000000  
25%          0.000000             1.000000  
50%          1.000000             1.000000  
75%          1.000000             1.000000  
max          1.000000             1.000000

#Feature Selection and Engineering

In [155]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 307 entries, 0 to 308
Data columns (total 16 columns):
 #   Column                 Non-Null Count  Dtype
---  ------                 --------------  -----
 0   AGE                    307 non-null    int64
 1   SMOKING                307 non-null    int64
 2   YELLOW_FINGERS         307 non-null    int64
 3   ANXIETY                307 non-null    int64
 4   PEER_PRESSURE          307 non-null    int64
 5   CHRONIC DISEASE        307 non-null    int64
 6   FATIGUE                307 non-null    int64
 7   ALLERGY                307 non-null    int64
 8   WHEEZING               307 non-null    int64
 9   ALCOHOL CONSUMING      307 non-null    int64
 10  COUGHING               307 non-null    int64
 11  SHORTNESS OF BREATH    307 non-null    int64
 12  SWALLOWING DIFFICULTY  307 non-null    int64
 13  CHEST PAIN             307 non-null    int64
 14  GENDER_ENCODED         307 non-null    int64
 15  LUNG_CANCER_ENCODED    307 non-null    int64


##Creating new features

In [156]:
# Social Behavior Metrics
social_behaviors = ['PEER_PRESSURE', 'ALCOHOL CONSUMING', 'ANXIETY']
df['SOCIAL_BEHAVIOR_SCORE'] = df[social_behaviors].sum(axis=1)

In [157]:
# Respiratory Health Metrics
respiratory_factors = ['ALLERGY', 'WHEEZING', 'COUGHING','SHORTNESS OF BREATH', 'CHEST PAIN', 'SWALLOWING DIFFICULTY']
df['RESPIRATORY_HEALTH_SCORE'] = df[respiratory_factors].sum(axis=1)

In [158]:
#Binning age
#May delete
bin_edges = [0, 18, 30, 40, 50, 60, 70, 80, np.inf]
bin_labels = ['0-17', '18-29', '30-39', '40-49', '50-59', '60-69', '70-79', '80+']
df['AGE_BINNED'] = pd.cut(df['AGE'], bins=bin_edges, labels=bin_labels, right=False)

In [159]:
from sklearn.preprocessing import StandardScaler, MinMaxScaler
minmax_scaler = MinMaxScaler()
#May delete
#Normalizing age
df['AGE_NORMALIZED'] = minmax_scaler.fit_transform(df[['AGE']])

In [182]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 307 entries, 0 to 308
Data columns (total 20 columns):
 #   Column                    Non-Null Count  Dtype   
---  ------                    --------------  -----   
 0   AGE                       307 non-null    int64   
 1   SMOKING                   307 non-null    int64   
 2   YELLOW_FINGERS            307 non-null    int64   
 3   ANXIETY                   307 non-null    int64   
 4   PEER_PRESSURE             307 non-null    int64   
 5   CHRONIC DISEASE           307 non-null    int64   
 6   FATIGUE                   307 non-null    int64   
 7   ALLERGY                   307 non-null    int64   
 8   WHEEZING                  307 non-null    int64   
 9   ALCOHOL CONSUMING         307 non-null    int64   
 10  COUGHING                  307 non-null    int64   
 11  SHORTNESS OF BREATH       307 non-null    int64   
 12  SWALLOWING DIFFICULTY     307 non-null    int64   
 13  CHEST PAIN                307 non-null    int64   
 14 

##Hypothesis Testing

In [160]:
from scipy.stats import ttest_ind
from scipy.stats import chi2_contingency

In [161]:
# Significance level
alpha = 0.01

In [181]:
lung_cancer_yes = df[df['LUNG_CANCER_ENCODED'] == 1]
lung_cancer_no = df[df['LUNG_CANCER_ENCODED'] == 0]

t_statistic, p_val = ttest_ind(lung_cancer_yes['AGE_NORMALIZED'], lung_cancer_no['AGE_NORMALIZED'])

print("Two-sample t-test results for 'Age' between individuals with and without lung cancer:")
print("T-statistic:", t_statistic)
print("P-value:", p_val)

if p_val < alpha:
    print("Age is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between age and lung cancer.")

Two-sample t-test results for 'Age' between individuals with and without lung cancer:
T-statistic: 0.9337358070462616
P-value: 0.3511788016726478
There is no significant association between age and lung cancer.


In [163]:
lung_cancer_yes = df[df['LUNG_CANCER_ENCODED'] == 1]
lung_cancer_no = df[df['LUNG_CANCER_ENCODED'] == 0]
t_statistic, p_val = ttest_ind(lung_cancer_yes['SOCIAL_BEHAVIOR_SCORE'], lung_cancer_no['SOCIAL_BEHAVIOR_SCORE'])

print("Two-sample t-test results for 'Social Behaviour Score' between individuals with and without lung cancer:")
print("T-statistic:", t_statistic)
print("P-value:", p_val)

if p_val < alpha:
    print("Social behaviour score is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between social behaviour score and lung cancer.")

Two-sample t-test results for 'Social Behaviour Score' between individuals with and without lung cancer:
T-statistic: 6.862724390434516
P-value: 3.7780481870627806e-11
Social behaviour score is a significant factor associated with lung cancer.


In [164]:
lung_cancer_yes = df[df['LUNG_CANCER_ENCODED'] == 1]
lung_cancer_no = df[df['LUNG_CANCER_ENCODED'] == 0]

t_statistic, p_val = ttest_ind(lung_cancer_yes['RESPIRATORY_HEALTH_SCORE'], lung_cancer_no['RESPIRATORY_HEALTH_SCORE'])

print("Two-sample t-test results for 'Respiratory Health Score' between individuals with and without lung cancer:")
print("T-statistic:", t_statistic)
print("P-value:", p_val)

if p_val < alpha:
    print("Respiratory health is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between respiratory health score and lung cancer.")

Two-sample t-test results for 'Respiratory Health Score' between individuals with and without lung cancer:
T-statistic: 8.951398148031423
P-value: 3.516793724852173e-17
Respiratory health is a significant factor associated with lung cancer.


In [165]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['SMOKING'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Smoking is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between smoking and lung cancer.")

Chi-square statistic: 1.03660258417898
P-value: 0.3086129394536048
Degrees of freedom: 1
There is no significant association between smoking and lung cancer.


In [166]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['YELLOW_FINGERS'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Yellow fingers is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between yellow fingers and lung cancer.")

Chi-square statistic: 8.16167567766154
P-value: 0.004278482282687588
Degrees of freedom: 1
Yellow fingers is a significant factor associated with lung cancer.


In [167]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['ANXIETY'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Anxiety is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between anxiety and lung cancer.")

Chi-square statistic: 5.172790511679675
P-value: 0.02294333209891496
Degrees of freedom: 1
There is no significant association between anxiety and lung cancer.


In [168]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['PEER_PRESSURE'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Peer pressure is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between peer presure and lung cancer.")

Chi-square statistic: 9.06380156265208
P-value: 0.0026071935790816752
Degrees of freedom: 1
Peer pressure is a significant factor associated with lung cancer.


In [169]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['CHRONIC DISEASE'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Chronic disease is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between chronic disease and lung cancer.")

Chi-square statistic: 3.7163086132041516
P-value: 0.05388337155409585
Degrees of freedom: 1
There is no significant association between chronic disease and lung cancer.


In [170]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['FATIGUE'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Fatigue is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between fatigue and lung cancer.")

Chi-square statistic: 6.663227182041708
P-value: 0.009842251628578925
Degrees of freedom: 1
Fatigue is a significant factor associated with lung cancer.


In [171]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['ALLERGY'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Allergy is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between allergy and lung cancer.")

Chi-square statistic: 33.25859693570448
P-value: 8.068227504465708e-09
Degrees of freedom: 1
Allergy is a significant factor associated with lung cancer.


In [172]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['WHEEZING'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Wheezing is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between wheezing and lung cancer.")

Chi-square statistic: 16.565226981436936
P-value: 4.700503225643931e-05
Degrees of freedom: 1
Wheezing is a significant factor associated with lung cancer.


In [173]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['ALCOHOL CONSUMING'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Alcohol consumption is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between alcohol consumption and lung cancer.")

Chi-square statistic: 23.181386698286833
P-value: 1.4741731209755625e-06
Degrees of freedom: 1
Alcohol consumption is a significant factor associated with lung cancer.


In [174]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['COUGHING'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Coughing is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between coughing and lung cancer.")

Chi-square statistic: 16.39539875571476
P-value: 5.140987577874612e-05
Degrees of freedom: 1
Coughing is a significant factor associated with lung cancer.


In [175]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['SHORTNESS OF BREATH'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Shortness of breath is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between shortness of breath and lung cancer.")

Chi-square statistic: 0.9915231589814771
P-value: 0.3193703858813325
Degrees of freedom: 1
There is no significant association between shortness of breath and lung cancer.


In [176]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['SWALLOWING DIFFICULTY'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Difficulty swallowing is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between difficulty swallowing and lung cancer.")

Chi-square statistic: 18.671843985405644
P-value: 1.5525827403348037e-05
Degrees of freedom: 1
Difficulty swallowing is a significant factor associated with lung cancer.


In [180]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['CHEST PAIN'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Chest pain is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between chest pain and lung cancer.")

Chi-square statistic: 9.141006265439408
P-value: 0.0024994360214342635
Degrees of freedom: 1
Chest pain is a significant factor associated with lung cancer.


In [178]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['GENDER_ENCODED'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Gender is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between gender and lung cancer.")

Chi-square statistic: 0.7848695559265448
P-value: 0.37565598163020864
Degrees of freedom: 1
There is no significant association between gender and lung cancer.


In [179]:
contingency_table = pd.crosstab(df['LUNG_CANCER_ENCODED'], df['AGE_BINNED'])
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_stat)
print("P-value:", p_val)
print("Degrees of freedom:", dof)

if p_val < alpha:
    print("Age group is a significant factor associated with lung cancer.")
else:
    print("There is no significant association between age group and lung cancer.")

Chi-square statistic: 7.983846891317463
P-value: 0.15712810681715061
Degrees of freedom: 5
There is no significant association between age group and lung cancer.


## Discussion

Significant Factors Associated with Lung Cancer:
* Yellow Fingers (P-value: 0.0043)
* Peer Pressure (P-value: 0.0026)
* Fatigue (P-value: 0.0098)
* Allergy (P-value: 8.07e-09)
* Wheezing (P-value: 4.70e-05)
* Alcohol Consuming (P-value: 1.47e-06)
* Coughing (P-value: 5.14e-05)
* Difficulty Swallowing (P-value: 1.55e-05)
* Chest Pain (P-value: 0.0025)
* Social Behavior Score (P-value: 3.78e-11)
* Respiratory Health Score (P-value: 3.52e-17)

No Significant Association with Lung Cancer:
* Chronic Disease (P-value: 0.0539)
* Shortness of Breath (P-value: 0.319)
* Gender (P-value: 0.376)
* Age Group (P-value: 0.157)
* Gender (P-value: 0.376)
* Smoking (P-value: 0.309)

Suggested Factors to use with model:
* Social Behaviour Score
* Respiratory Health Score
* Wheezing
* Alcohol Consuming

(Subject to change)